src/ctypes/ldouble.ml

```diff
@@ -0,0 +1,121 @@
+(*
+ * Copyright (c) 2016 Andy Ray.
+ *
+ * This file is distributed under the terms of the MIT License.
+ * See the file LICENSE for details.
+ *)
+
+external init : unit -> unit = "ldouble_init"
+let () = init ()
+
+type t
+
+external to_float : t -> float = "ctypes_ldouble_to_float"
+external of_float : float -> t = "ctypes_ldouble_of_float"
+external to_int : t -> int = "ctypes_ldouble_to_int"
+external of_int : int -> t = "ctypes_ldouble_of_int"
+
+external format : string -> t -> string = "ctypes_ldouble_format"
+let to_string d = format "%Lf" d
```

Comment:
Is there a need to pass in the format string from OCaml here?  It seems better to hard-code it in the C stub.